index.js

```diff
@@ -265,3 +127,38 @@ function encrypt (block, nonce, key) {
 function decrypt (block, nonce, key) {
   return encrypt(block, nonce, key) // symmetric
 }
+
+function blockhash (block, padding, hashKey) {
+  sodium.crypto_generichash(hash, block, hashKey)
+  padding.set(hash.subarray(0, 8)) // copy first 8 bytes of hash
+  hash.fill(0) // clear nonce buffer
+}
+
+function encryptBlock (index, block, id, blockKey, hashKey) {
```

Comment:
Probably need a walk through of this to fully understand why it differs from the default encryption.

But immediately wondering whether the hypercore's `fork` is accounted for. Was the `DefaultEncryption`'s using the fork in the padding replaced with hashing the block with the hashKey with the encryption key id at the end?